### PR TITLE
Change some entity, complete the CheckProfile Usecase.

### PR DESCRIPTION
**Pull request by KunLW**

### CODE REVIEW

**Review by tlan4 — DISMISSED** [review]
Nice, role factory is very useful, but I think you should rename my_dao to something more descriptive

**Review by GitHaoBao — APPROVED** [review]

**Comment by KunLW** [comment]
> Nice, role factory is very useful, but I think you should rename my_dao to something more descriptive

This is just a template DAO for my code to compile. I will delete it in the future.

**Review by staceylry — DISMISSED** [review]

**Review by tlan4 — APPROVED** [review]

**Review by jianwang0722 — APPROVED** [review]

File: src/main/java/check_profile_validation/CheckProfileIGateway.java [hunk 1]
```diff
@@ -11,15 +11,15 @@
 import java.util.UUID;
 
 public interface CheckProfileIGateway {
-    List<Role> getUserRolesByUid(Integer target);
+//    List<Role> getUserRolesByUid(Integer target);
 
     User getUserByUid(Integer target);
 
-    PropertyChangeSupport getObservable();
```
Comment (KunLW) [inline src/main/java/check_profile_validation/CheckProfileIGateway.java:18]: Make gateway as a observable thing

**Review by kaizen-w — APPROVED** [review]
Looks good.

**Pull request merged**